### PR TITLE
example/dice: Do not use semconv

### PR DESCRIPTION
**Pull request by pellared**
## Why

Per https://github.com/open-telemetry/opentelemetry-go/pull/4846#issuecomment-1904656729

Supersedes https://github.com/open-telemetry/opentelemetry-go/pull/4846

## What

Use simply `resource.Default()` and set `OTEL_RESOURCE_ATTRIBUTES` instead of using `resource.NewWithAttributes` to mitigate the "resource merge conflict" runtime error that the users encounter have when bumping OTel Go when they are using the existing "getting started" example (https://opentelemetry.io/docs/languages/go/getting-started/#initialize-the-opentelemetry-sdk).

A PR to update https://opentelemetry.io/docs/languages/net/automatic/getting-started/ will be created as a follow-up to this PR.

Related to https://github.com/open-telemetry/opentelemetry-go/issues/2341

PTAL @chalin @svrnm

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

File: example/dice/otel.go [hunk 1]
```diff
@@ -83,12 +78,8 @@ func setupOTelSDK(ctx context.Context, serviceName, serviceVersion string) (shut
 	return
 }
 
-func newResource(serviceName, serviceVersion string) (*resource.Resource, error) {
-	return resource.Merge(resource.Default(),
-		resource.NewWithAttributes(semconv.SchemaURL,
-			semconv.ServiceName(serviceName),
-			semconv.ServiceVersion(serviceVersion),
-		))
+func newResource() *resource.Resource {
```
Comment (dashpole) [inline example/dice/otel.go:81]: Isn't this used by default?  Should we remove resource setting from the example entirely?

Comment (pellared) [reply example/dice/otel.go:81]: Sorry I got you wrong. I see you mean completely remove setting of the resource... Makes sense. I will do that unless anyone disagrees.

@MrAlias @dmathieu are you OK with removing the code that sets the resource?

Comment (MrAlias) [reply example/dice/otel.go:81]: > @MrAlias @dmathieu Are you OK with removing the code that sets the resource?

SGTM

Comment (pellared) [reply example/dice/otel.go]: Addressed eb1decfbc677d43d84daa017ba0eeece56c6962f

**Review by dashpole — APPROVED** [review]

**Pull request merged**